libflux/src/core/parser/mod.rs

```diff
@@ -577,6 +577,76 @@ impl Parser {
             MonoType::Invalid
         }
     }
+
+    // Record = "{" [ Identifier (Suffix1 | Suffix2) ] "}"
+    // Suffix1 = ":" MonoType { "," Property }
+    // Suffix2 = "with" [Properties]
+    #[cfg(test)]
+    fn parse_record(&mut self) -> Record {
+        let start = self.open(TOK_LBRACE, TOK_RBRACE);
+        let mut properties: Option<Vec<PropertyType>> = None;
+        let property: PropertyType;
+
+        let mut id: Option<Identifier> = None;
+        let _id;
+        let t = self.peek();
+        if t.tok == TOK_IDENT {
+            // Indentifier
+            _id = self.parse_identifier();
+            // suffix one needs attention
+            let t2 = self.peek();
+            if t2.tok == TOK_COLON {
+                let mut ps = Vec::<PropertyType>::new();
+                self.consume(); // :
```

Comment:
Can you use `self.expect` here for edge cases?

Reply:
✅

Reply:
Oops, you already check for a `:` before consuming. My bad.